internal/nginx/config/servers.go

```diff
@@ -75,23 +75,12 @@ func createServer(virtualServer dataplane.VirtualServer) http.Server {
 }
 
 func createLocations(pathRules []dataplane.PathRule, listenerPort int32) []http.Location {
```

Comment:
this case looks like a bug.

yaml:
```yaml
apiVersion: gateway.networking.k8s.io/v1beta1
kind: HTTPRoute
metadata:
  name: coffee
spec:
  parentRefs:
  - name: gateway
    sectionName: http
  hostnames:
  - "cafe.example.com"
  rules:
  - matches:
    - path:
        type: Exact
        value: /coffee
    backendRefs:
    - name: tea
      port: 80
  - matches:
    - path:
        type: PathPrefix
        value: /coffee/
    backendRefs:
    - name: tea
      port: 80
  - matches:
    - path:
        type: PathPrefix
        value: /coffee
    backendRefs:
    - name: coffee
      port: 80
```

NGINX:
```nginx
    location = /coffee {

        proxy_set_header Host $gw_api_compliant_host;
        proxy_pass http://default_tea_80$request_uri;
    }

    location /coffee {  # is this location a bug?

        proxy_set_header Host $gw_api_compliant_host;
        proxy_pass http://default_coffee_80$request_uri;
    }

    location /coffee/ {

        proxy_set_header Host $gw_api_compliant_host;
        proxy_pass http://default_tea_80$request_uri;
    }
```

Reply:
This looks correct based on the configured YAML.

Reply:
Does this go against Gateway API spec though?
```
curl --resolve cafe.example.com:$GW_PORT:$GW_IP http://cafe.example.com:$GW_PORT/coffeeabc
Server address: 10.244.0.9:8080
Server name: coffee-7dd75bc79b-hpq6d
Date: 10/Jul/2023:20:55:08 +0000
URI: /coffeeabc
Request ID: a9df3294f2787ce1b0561af398041d95
```

Reply:
Oh I see....so we'd essentially be deleting the `/coffee Prefix` option from their configuration in order to create a proper config.